test/rbac/gettest.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rbac
+
+import (
+	"context"
+	"github.com/onosproject/onos-config/test/utils/gnmi"
+	"github.com/onosproject/onos-config/test/utils/proto"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+const (
+	interfaceName    = "starbucks"
+	rootPath         = "/interfaces/interface[name=" + interfaceName + "]"
+	namePath         = rootPath + "/config/name"
+	enabledPath      = rootPath + "/config/enabled"
+	descriptionPath  = rootPath + "/config/description"
+	nameValue        = "starbucks"
+	descriptionValue = "description"
+)
+
+func setUpInterfaces(t *testing.T, target string, password string) {
+	// get an access token
+	token, err := fetchATokenViaKeyCloak("https://keycloak-dev.onlab.us/auth/realms/master", "alicea", password)
+	assert.NoError(t, err)
+	assert.NotNil(t, token)
+
+	// Make a GNMI client to use for requests
+	ctx := getContext(context.Background(), token)
+	gnmiClient := gnmi.GetGNMIClientWithContextOrFail(ctx, t)
+
+	// Create interface tree using gNMI client
+	setNamePath := []proto.DevicePath{
+		{DeviceName: target, Path: namePath, PathDataValue: nameValue, PathDataType: proto.StringVal},
+	}
+	gnmi.SetGNMIValueWithContextOrFail(ctx, t, gnmiClient, setNamePath, gnmi.NoPaths, gnmi.NoExtensions)
+
+	// Set initial values for Enabled and Description using gNMI client
+	setInitialValuesPath := []proto.DevicePath{
+		{DeviceName: target, Path: enabledPath, PathDataValue: "true", PathDataType: proto.BoolVal},
+		{DeviceName: target, Path: descriptionPath, PathDataValue: descriptionValue, PathDataType: proto.StringVal},
+	}
+	gnmi.SetGNMIValueWithContextOrFail(ctx, t, gnmiClient, setInitialValuesPath, gnmi.NoPaths, gnmi.NoExtensions)
+}
+
+// TestGetOperations tests get operations to a protected API with various users
+func (s *TestSuite) TestGetOperations(t *testing.T) {
+	testCases := map[string]struct {
+		username      string
+		expectedValue string
+	}{
+		"Aether ROC Admin": {
```

Comment:
I was thinking - if you added 3 interfaces `acme`, `starbucks` and `other` then you could test that

- alicea could see all 3 interfaces
- daisyd and elmerf could only `starbucks`
- fredf and gandalfg could only see `acme`
- bobc could see none

This would cover more scenarios - you could leave to a separate test too.

Reply:
good ideas. I should be able to add them here pretty easily since this is already table driven.